### ISSUE TITLE
Assertion `VY != 0 && "This should have been folded"` failed

### ISSUE THREAD

**Issue by chfast** [llvm/llvm-project#63406]
```
llc: /root/llvm-project/llvm/lib/Target/Hexagon/HexagonISelLowering.cpp:3256: llvm::SDValue llvm::HexagonTargetLowering::LowerUAddSubO(llvm::SDValue, llvm::SelectionDAG&) const: Assertion `VY != 0 && "This should have been folded"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /opt/compiler-explorer/clang-assertions-trunk/bin/llc -o /app/output.s -x86-asm-syntax=intel <source>
1.	Running pass 'Function Pass Manager' on module '<source>'.
2.	Running pass 'Hexagon DAG->DAG Pattern Instruction Selection' on function '@f'
 #0 0x000055f73f95645f llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x357845f)
 #1 0x000055f73f953bb4 SignalHandler(int) Signals.cpp:0:0
 #2 0x00007f06d10fc420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #3 0x00007f06d0bc900b raise (/lib/x86_64-linux-gnu/libc.so.6+0x4300b)
 #4 0x00007f06d0ba8859 abort (/lib/x86_64-linux-gnu/libc.so.6+0x22859)
 #5 0x00007f06d0ba8729 (/lib/x86_64-linux-gnu/libc.so.6+0x22729)
 #6 0x00007f06d0bb9fd6 (/lib/x86_64-linux-gnu/libc.so.6+0x33fd6)
 #7 0x000055f73d8b44b2 llvm::HexagonTargetLowering::LowerUAddSubO(llvm::SDValue, llvm::SelectionDAG&) const (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x14d64b2)
 #8 0x000055f73f61039c (anonymous namespace)::SelectionDAGLegalize::LegalizeOp(llvm::SDNode*) (.part.0) LegalizeDAG.cpp:0:0
 #9 0x000055f73f61433c llvm::SelectionDAG::Legalize() (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x323633c)
#10 0x000055f73f71f23c llvm::SelectionDAGISel::CodeGenAndEmitDAG() (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x334123c)
#11 0x000055f73f722a18 llvm::SelectionDAGISel::SelectAllBasicBlocks(llvm::Function const&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x3344a18)
#12 0x000055f73f724712 llvm::SelectionDAGISel::runOnMachineFunction(llvm::MachineFunction&) (.part.0) SelectionDAGISel.cpp:0:0
#13 0x000055f73d885f6a llvm::HexagonDAGToDAGISel::runOnMachineFunction(llvm::MachineFunction&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x14a7f6a)
#14 0x000055f73ebe601e llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (.part.0) MachineFunctionPass.cpp:0:0
#15 0x000055f73f17c071 llvm::FPPassManager::runOnFunction(llvm::Function&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x2d9e071)
#16 0x000055f73f17c2b9 llvm::FPPassManager::runOnModule(llvm::Module&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x2d9e2b9)
#17 0x000055f73f17cb32 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x2d9eb32)
#18 0x000055f73cc66734 compileModule(char**, llvm::LLVMContext&) llc.cpp:0:0
```

```llvm
target triple = "hexagon-unknown-linux-gnu"

define i64 @f(i64 %x, ptr %p) {
  %s = add i64 -9223372036854775808, %x
  %c = icmp ugt i64 -9223372036854775808, %s
  store i1 %c, ptr %p
  ret i64 %s
}
```

https://godbolt.org/z/jsz3Pn3e4

**Comment by kparzysz-quic** [comment]
@chfast: Thanks for the bug reports!

**Comment by chfast** [comment]
To explain myself, I started running `llvm-isel-fuzzer` as a part of a DAGCombiner patch review.

After some of your fixes to Hexagon, I can run the fuzzer for this target longer, so good job. But don't treat these bugs as priority. You can also drop me some additional triples or flags for llc to test.

**Comment by kparzysz-quic** [comment]
Fixed in dbc283bb9ec0392624458edf967c00877adb0bec.
